t.go

```diff
@@ -21,3 +23,87 @@ type T interface {
 	Skipf(format string, args ...interface{})
 	Skipped() bool
 }
+
+// RunT brings in more of the *testing.T API including Run and Parallel which are needed
+// for matrix tests.
+// *testing.T satisfies the RunT interface.
+type RunT interface {
+	T
+	Run(string, func(*testing.T)) bool
+	Parallel()
+}
+
+// NewTestRunner creates a test runner that works with matrix testing by
+// upgrading a T to a RunT.
+// This is useful for converting T types to work with matrix testing functions.
+// If the concrete type underlying T doesn't implement Run then the
+// test will fail if Run is called. If the underlying type doesn't implement
+// Parallel, then the test won't be parallel.
+func NewTestRunner(t T) RunT {
+	if runT, ok := t.(RunT); ok {
+		return runT
+	}
+	return simpleRunT{T: t, orig: t}
+}
+
+// simpleRunT provides RunT functionality for plain T types
+type simpleRunT struct {
+	T
+	orig T
+}
+
+func (s simpleRunT) Run(name string, f func(*testing.T)) bool {
+	if runT, ok := s.orig.(RunT); ok {
+		return runT.Run(name, f)
+	}
+	//nolint:staticcheck // QF1008: could remove embedded field "T" from selector
+	s.T.Logf("Run not supported by %T", s.orig)
+	//nolint:staticcheck // QF1008: could remove embedded field "T" from selector
+	s.T.FailNow()
+	return false
+}
+
+func (s simpleRunT) Parallel() {
+	if parallel, ok := s.orig.(interface{ Parallel() }); ok {
+		parallel.Parallel()
+	}
+}
+
+// runTHelper is a tiny wrapper around T that provides Fail and Parallel methods
+// by casting to appropriate interfaces. This can be embedded since it's not parameterized.
+type runTHelper struct {
+	T
+}
+
+func (r runTHelper) Fail() {
+	if failer, ok := r.T.(interface{ Fail() }); ok {
+		failer.Fail()
+		return
+	}
+	// Fallback
+	//nolint:staticcheck // QF1008: could remove embedded field "T" from selector
+	r.T.FailNow()
+}
+
+func (r runTHelper) Parallel() {
+	if parallel, ok := r.T.(interface{ Parallel() }); ok {
+		parallel.Parallel()
+		return
+	}
+	// If not supported, we just continue - parallel is optional
+}
+
+// RunWithReWrap is a helper that runs a subtest and automatically handles ReWrap logic.
+// This should be used instead of calling t.Run in tests that use
+// ReplaceLogger, BufferedLogger, or ExtraDetailLogger.
+func RunWithReWrap(t RunT, name string, f func(RunT)) bool {
```

Comment:
What does the name "RunWithReWrap" convey to the caller of this function?  It feels like the need to "re-wrap" is a detail internal to this package.  What is the caller getting?  Is it just a test run with the logs prefixed?